rpcserver.go

```diff
@@ -6743,6 +6747,38 @@ func (r *rpcServer) ListPermissions(_ context.Context,
 	}, nil
 }
 
+// CheckMacaroonPermissions checks the caveats and permissions of a macaroon.
+func (r *rpcServer) CheckMacaroonPermissions(ctx context.Context,
+	req *lnrpc.CheckMacPermRequest) (*lnrpc.CheckMacPermResponse, error) {
+
+	// Turn grpc macaroon permission into bakery.Op for the server to
+	// process.
+	permissions := make([]bakery.Op, 0)
+	for _, perm := range req.Permissions {
+		newPerm := bakery.Op{
+			Entity: perm.Entity,
+			Action: perm.Action,
+		}
+
+		permissions = append(permissions, newPerm)
+	}
+
+	err := r.macService.CheckMacAuth(
```

Comment:
Surely we want to be able to check macaroon perms for fullMethod uri names other than for `/lnrpc.Lightning/CheckMacPermissions`? like, shouldn't the fullMethod argument for the CheckMacAuth come from the rpc request as another member of `CheckMacPermRequest`?

Reply:
Oh dang, great catch! I think I put that in as a placeholder and forgot to change